### PR TITLE
Add nonce to render async call WIP

### LINKED ISSUE #1140: Update render_async gem so reports load, hopefully
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**

Get reports to render asynchronously, with nonces!

* **What feature or behavior is this required for?**

* **How could we solve this issue? (Not knowing is okay!)**

Run `bundle update render_async` and add `nonce: content_security_policy_script_nonce` to the render_async stuff in `app/views/reports/index.html.erb`. (The nonce is required so it's whitelisted by our content security policy.)

* **Anything else?**

See #1095 , #1009

### PR DESCRIPTION
**Pull request by tingaloo**
I rule and have completed some work on Case Manager that's ready for review!

Add nonces to render async to bypass CSP.

This pull request makes the following changes:
* Add nonces
* update render async gem

It relates to the following issue #s: 
* Fixes #1140

### CODE REVIEW

File: app/views/reports/index.html.erb [hunk 1]
```diff
@@ -8,6 +8,6 @@
   <div class="progress-bar progress-bar-striped"</div>
 </div>
  -->
-<%= render_async patients_report_path(timeframe: 'weekly') %>
-<%= render_async patients_report_path(timeframe: 'monthly') %>
-<%= render_async patients_report_path(timeframe: 'yearly') %>
+<%= render_async patients_report_path(timeframe: 'weekly', nonce: content_security_policy_script_nonce) %>
```
Comment (colinxfleming) [inline app/views/reports/index.html.erb]: ah, I see. Nonce is an options argument of `render_async`, not `patients_report_path`, so pop it outside of the parens and it should blend. e.g. `render_async patients_report_path(...), nonce: ...`

**Comment by colinxfleming** [comment]
@tingaloo , sorry, I'm a dingdong. The version of the gem with our changes in them is 0.2.3. Should work as is once we have that in place. If you have a chance feel free to rack this change, otherwise I'll do it when I have some quiet time over the next few days. sorry for the scrub directions on my part!

**Comment by tingaloo** [comment]
Updated gemfile, still not showing.

```Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' 'unsafe-eval'". Either the 'unsafe-inline' keyword, a hash ('sha256-+PC+Kf6qxwFk9MeuRMDmsALBNWNX473zplWkiVYhJgY='), or a nonce ('nonce-...') is required to enable inline execution.```

Here is the updated js
```<script nonce="3AmhCU3l94wsnEjXKSCnzhR2B4jEXMRojWkHpSHEOKs=">
//<![CDATA[

    (function($){
      $.ajax({
          url: "/reports/weekly",
        })
        .done(function(response, status) {
          $("#render_async_3629e070061500564440").html(response);
        })
        .fail(function(response, status) {
          $("#render_async_3629e070061500564440").html(response);
        })
        .always(function(response, status) {
          $("#render_async_3629e070061500564440_spinner").hide();
        });
    }(jQuery));

//]]>
</script>```

**Comment by colinxfleming** [comment]
That is super weird, this PR is working for me! 

screen shot 2017-07-21 at 11 36 31 pm

The violation up there is, I think, complaining about `gon` and not an actual violation. I'll submit a fix to get that to hush since it is pretty ugmo.

**Comment by tingaloo** [comment]
I guess my browser is caching or something. Merge on!

**Pull request merged**